backend/app/api/endpoints/prompt_optimization.py

```
@@ -0,0 +1,37 @@
+    ApplyPromptChangesRequest,
+    ApplyPromptChangesResponse,
+)
+from app.services.prompt_optimization import apply_prompt_changes
+
+router = APIRouter()
+
+
+@router.post("/apply", response_model=ApplyPromptChangesResponse)
+def apply_prompt_optimization_changes(
+    request: ApplyPromptChangesRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)